services/src/main/java/org/apache/unomi/services/actions/ActionExecutorDispatcher.java

```diff
@@ -124,13 +126,21 @@ public Action getContextualAction(Action action, Event event) {
             if (value instanceof String) {
                 String s = (String) value;
                 try {
-                    // check if we have special values
-                    if (s.contains(VALUE_NAME_SEPARATOR)) {
-                        final String valueType = StringUtils.substringBefore(s, VALUE_NAME_SEPARATOR);
-                        final String valueAsString = StringUtils.substringAfter(s, VALUE_NAME_SEPARATOR);
-                        final ValueExtractor extractor = valueExtractors.get(valueType);
-                        if (extractor != null) {
-                            value = extractor.extract(valueAsString, event);
+                    if (s.contains(PLACEHOLDER_PREFIX)) {
+                        while (s.contains(PLACEHOLDER_PREFIX)) {
+                            String substring = s.substring(s.indexOf(PLACEHOLDER_PREFIX) + 2, s.indexOf(PLACEHOLDER_SUFFIX));
```

Comment:
What happens if the PLACEHOLDER_SUFFIX is not found, does it raise an exception because indexOf is < 0 ?